### PR TITLE
Revise kubernetes documentation

### LINKED ISSUE #1003: kubernetes doc
**Describe the error/improvement to an existing document or image**

some feedback: https://github.com/clearlinux/distribution/issues/1663#issuecomment-577411817

### PR DESCRIPTION
**Pull request by puneetse**
This PR is a major update and revision to the order of concepts and consolidating commands to make the document more readable. There are still known complexities and issues with the process over all. This PR is not meant to address all issues or be the final set of instructions. It is meant to take a step in the direction we want to go.

Most notably it:
- Presents Kubernetes concepts in order of understanding with reference material pulled out and away from the main tutorial. 
- Adds the cloud-native-setup scripts as the first option since they have been bundled in Clear Linux.
- Adds docker and containerd options for container runtimes

Related to: https://github.com/clearlinux/clear-linux-documentation/issues/894
Related to: https://github.com/clearlinux/clear-linux-documentation/issues/1003

Closes #1003

### CODE REVIEW

File: source/tutorials/kubernetes.rst [hunk 1]
```diff
@@ -3,354 +3,657 @@
 Kubernetes\*
 ############
 
-This tutorial describes how to install, configure, and start the
-`Kubernetes container orchestration system`_ on |CL-ATTR| using CRI+O and kata-runtime.
+This tutorial describes how to install, configure, and start the `Kubernetes
+container orchestration system`_ on |CL-ATTR|.
+
+A Kubernetes cluster can be setup on |CL| using scripts to automated the
+process or manually step-by-step.
```
Comment (eadamsintel) [inline source/tutorials/kubernetes.rst]: automated should be automate.  I would rephrase to say "A kubernetes cluster can be setup on CL using the CL cloud-native-setup scripts to automate the process or can be setup through a manual step by step process.  This tutorial covers both scenarios."  I would have in page hyperlinks to the section for the scripts and the section for manual process.  I don't just like saying "scripts" without any context on where they came from so I think adding in they are Clear Linux cloud-native-setup scripts gives more confidence this isn't just some unmaintained script somewhere.

File: source/tutorials/kubernetes.rst [hunk 2]
```diff
@@ -3,354 +3,657 @@
...
-   sudo swupd bundle-add cloud-native-basic
+
+Set up Kubernetes automatically
+*******************************
+
+|CL| provides `cloud-native-setup scripts`_ to automate system setup and
+Kubernetes cluster initialization which allows you to get a cluster up and
+running quickly.
 
 .. note::
```
Comment (eadamsintel) [inline source/tutorials/kubernetes.rst]: This did not render properly in my view.

File: source/tutorials/kubernetes.rst [hunk 3]
```diff
@@ -3,354 +3,657 @@
...
 
-   For more on networking plugins, see `Install pod network add-on`_.
+   By default, the :file:`create_stack.sh` script setups Kubernetes with
+   **canal for container networking** and **crio for container runtime**.
+   The script can be configured to use other CNI and CRI by following the
```
Comment (eadamsintel) [inline source/tutorials/kubernetes.rst]: CNI's and CRI's

File: source/tutorials/kubernetes.rst [hunk 4]
```diff
@@ -3,354 +3,657 @@
...
+   configure them now because the scripts will propagate proxy configuration
+   based on the running configuration. See `Proxy issues`_ for details.
+
+
+#. Run the :file:`system-setup.sh` script to configure the |CL| system
+   settings.
+
+   .. code-block:: bash
+
+      sudo /usr/share/clr-k8s-examples/setup_system.sh
```
Comment (eadamsintel) [inline source/tutorials/kubernetes.rst]: I never had to use sudo

File: source/tutorials/kubernetes.rst [hunk 5]
```diff
@@ -3,354 +3,657 @@
...
+
+#. Run the :file:`create_stack.sh` script to initialize the Kubernetes node.
+   and setup a container network plugin
+
+   .. code-block:: bash
+
+      sudo /usr/share/clr-k8s-examples/create_stack.sh
```
Comment (eadamsintel) [inline source/tutorials/kubernetes.rst]: I didn't have to use sudo

File: source/tutorials/kubernetes.rst [hunk 6]
```diff
@@ -3,354 +3,657 @@
...
+
+#. Follow the output on the screen continue onto the section on `using your
```
Comment (eadamsintel) [inline source/tutorials/kubernetes.rst]: .... to continue onto ...

File: source/tutorials/kubernetes.rst [hunk 7]
```diff
@@ -3,354 +3,657 @@
...
+
+
+#. Run the :file:`system-setup.sh` script to configure the |CL| system
+   settings.
+
+   .. code-block:: bash
+
+      sudo /usr/share/clr-k8s-examples/setup_system.sh
+
+#. Run the :file:`create_stack.sh` script to initialize the Kubernetes node.
```
Comment (eadamsintel) [inline source/tutorials/kubernetes.rst]: no period after node

File: source/tutorials/kubernetes.rst [hunk 8]
```diff
@@ -3,354 +3,657 @@
...
+   cluster <Use your cluster>`_.
+
+
+Set up Kubernetes manually
+**************************
+
+Configure host system
+=====================
+
+This tutorial uses the basic default Kubernetes configuration for to get
```
Comment (eadamsintel) [inline source/tutorials/kubernetes.rst]: don't need "for"

File: source/tutorials/kubernetes.rst [hunk 9]
```diff
@@ -3,354 +3,657 @@
...
+   directions on the `README
+   <https://github.com/clearlinux/cloud-native-setup/blob/master/clr-k8s-examples/README.md>`_.
+   
+   See `What is a Container Network Interface (CNI)?`_ and `What is a
+   Container Runtime Interface (CRI)?`_ for more information.
+
+.. important::
+
+   If any network proxy settings are required for Internet connectivity
+   configure them now because the scripts will propagate proxy configuration
```
Comment (eadamsintel) [inline source/tutorials/kubernetes.rst]: I would recommend that people put their proxy info in /etc/profile.d/proxy.sh as that is one place where the script looks for proxy information.  The /etc/environment is the other place but I would recommend they pick one. I always put in /etc/profile.d/proxy.sh

File: source/tutorials/kubernetes.rst [hunk 10]
```diff
@@ -3,354 +3,657 @@
...
-.. code-block:: console
+#. Once the cluster initialization is complete, continue reading about how to
+   `Use your cluster`_.
 
-    [crio.runtime]
-    manage_network_ns_lifecycle = true
 
-Use your cluster
-****************
+CRI+O
```
Comment (eadamsintel) [inline source/tutorials/kubernetes.rst]: You should do crio first, containerd second, and docker last.

File: source/tutorials/kubernetes.rst [hunk 11]
```diff
@@ -3,354 +3,657 @@
...
+      
+      In this example, the :command:`--pod-network-cidr 10.244.0.0/16`
+      parameter is fto use *flannel* as the pod networking. See `Choose a pod
+      network add-on`_ for more information.
 
-*  Deploying Kubernetes with a `cloud-native-setup`_
+   .. code-block:: bash
 
-*  :ref:`Kubernetes best practices <kubernetes-bp>`
+      sudo kubeadm init \
```
Comment (eadamsintel) [inline source/tutorials/kubernetes.rst]: Did this work for you? This is where I lost a few days trying to figure out my problem.  I had to to this for crio to work and probably also would have to do for containerd.  I am not sure about docker though.  Also, you don't have to specify the cri-socket if only one is running.  It is probably good to do anyways.  

```You need to set the kubelet to use system as the cgroup. This can be set in kubeadm.yaml as well which is more convenient then reloading systemd daemon
sudo mkdir -p /etc/systemd/system/kubelet.service.d/
cat << EOF | sudo tee  /etc/systemd/system/kubelet.service.d/10-cgroup-driver.conf
[Service]
Environment="KUBELET_EXTRA_ARGS=--cgroup-driver=systemd"
EOF
sudo systemctl daemon-reload
```

File: source/tutorials/kubernetes.rst [hunk 12]
```diff
@@ -3,354 +3,657 @@
...
+
+      sudo /usr/share/clr-k8s-examples/setup_system.sh
+
+#. Run the :file:`create_stack.sh` script to initialize the Kubernetes node.
+   and setup a container network plugin
+
+   .. code-block:: bash
+
+      sudo /usr/share/clr-k8s-examples/create_stack.sh
+
```
Comment (eadamsintel) [inline source/tutorials/kubernetes.rst]: You should have the instructions for  reset_stack.sh in here as well.   It would be worth mentioning that if it complains about multiple cri sockets to just use `sudo systemctl stop ....` whichever cri that is not needed.

File: source/tutorials/kubernetes.rst [hunk 13]
```diff
@@ -3,354 +3,657 @@
...
-       exit
+   .. important:: 
 
-Configure and run Kubernetes
-****************************
+      You may need to add additional parameters to the command below,
+      depending the pod network addon in use. 
+      
+      In this example, the :command:`--pod-network-cidr 10.244.0.0/16`
+      parameter is fto use *flannel* as the pod networking. See `Choose a pod
```
Comment (eadamsintel) [inline source/tutorials/kubernetes.rst]: fto = to

File: source/tutorials/kubernetes.rst [hunk 14]
```diff
@@ -3,354 +3,657 @@
...
+
+#. Run the :file:`create_stack.sh` script to initialize the Kubernetes node.
+   and setup a container network plugin
+
+   .. code-block:: bash
+
+      sudo /usr/share/clr-k8s-examples/create_stack.sh
+
+#. Follow the output on the screen continue onto the section on `using your
+   cluster <Use your cluster>`_.
```
Comment (eadamsintel) [inline source/tutorials/kubernetes.rst]: You should consider adding a kata section here as well. Kata is deployed differently and will automatically setup runtimes for kata-qemu (equivalent to kata-runtime in the manual instructions), kata-nemu, and kata-fc.  You have to request kata containers differently in the yaml based on this.

File: source/tutorials/kubernetes.rst [hunk 15]
```diff
@@ -3,354 +3,657 @@
...
-#.  Register kata-runtime as a `RuntimeClass handler`_:
+      sudo mkdir -p  /etc/systemd/system/kubelet.service.d/
 
-    .. code-block:: bash
+      cat << EOF | sudo tee  /etc/systemd/system/kubelet.service.d/0-containerd.conf
+      [Service]                                                 
+      Environment="KUBELET_EXTRA_ARGS=--container-runtime=remote --runtime-request-timeout=15m --container-runtime-endpoint=unix:///run/containerd/containerd.sock"
+      EOF
 
-       cat << EOF | kubectl apply -f -
```
Comment (eadamsintel) [inline source/tutorials/kubernetes.rst]: It really would be better if we had a way to refer to the kata github for this.  It will be hard to keep this in sync.  The one ont he github is https://github.com/kata-containers/packaging/blob/master/kata-deploy/k8s-1.14/kata-qemu-runtimeClass.yaml but that refers to kata-qemu since it is part of kata-deploy.   This is why I think the setup scripts and these instructions should do the same thing.   Maybe, instead of using kata-containers in containers-virt your instructions should just instruct to use kata-deploy.  See https://github.com/kata-containers/packaging/tree/5717e6cbe77e370fd725919bd968dfb9582dea41/kata-deploy for more details.  They recommend kubectl apply -f https://raw.githubusercontent.com/kata-containers/packaging/master/kata-deploy/k8s-1.14/kata-qemu-runtimeClass.yaml  but I would actually look at the create_stack.sh script and follow the same process they use which sets up all the kata runtimes.  These instructions will be valid for longer if you do it that way.

File: source/tutorials/kubernetes.rst (hunk 2 above)
Comment (puneetse) [reply source/tutorials/kubernetes.rst]: You have to build the docs locally to the documentation render properly. The GH render is a best-effort.

File: source/tutorials/kubernetes.rst (hunk 9 above)
Comment (puneetse) [reply source/tutorials/kubernetes.rst]: I'm not going to address this in this PR. The linked proxy documentation does explain this.

File: source/tutorials/kubernetes.rst (hunk 11 above)
Comment (puneetse) [reply source/tutorials/kubernetes.rst]: I had the same issues as you. I'm not going to address it in this PR. See from OP
> There are still known complexities and issues with the process over all. This PR is not meant to address all issues or be the final set of instructions.

File: source/tutorials/kubernetes.rst (hunk 14 above)
Comment (puneetse) [reply source/tutorials/kubernetes.rst]: I'm not going to address it in this PR. See from OP
> There are still known complexities and issues with the process over all. This PR is not meant to address all issues or be the final set of instructions.

File: source/tutorials/kubernetes.rst (hunk 15 above)
Comment (puneetse) [reply source/tutorials/kubernetes.rst]: I'm not going to address it in this PR. See from OP
> There are still known complexities and issues with the process over all. This PR is not meant to address all issues or be the final set of instructions.

File: source/tutorials/kubernetes.rst (hunk 11 above)
Comment (puneetse) [reply source/tutorials/kubernetes.rst]: Actually, I added this now. I remember testing it.

File: source/tutorials/kubernetes.rst [hunk 16]
```diff
@@ -3,354 +3,716 @@
...
+   runtime. The script can be configured to use other CNI's and CRI's by
+   following the directions on the `README
+   <https://github.com/clearlinux/cloud-native-setup/blob/master/clr-k8s-examples/README.md>`_.
+   
+   See `What is a Container Network Interface (CNI)?`_ and `What is a
+   Container Runtime Interface (CRI)?`_ for more information.
+
+.. important::
+
+   If network proxy settings are required for Internet connectivity, configure
```
Comment (eadamsintel) [inline source/tutorials/kubernetes.rst]: It might be worth stating that the script looks at /etc/environment and /etc/profile.d/proxy.sh for proxy information.   I've seen guides state to put proxy info in .bashrc  or /etc/bash.bashrc and the script doesn't check there.

File: source/tutorials/kubernetes.rst [hunk 17]
```diff
@@ -3,354 +3,716 @@
...
+
+      sudo systemctl stop docker
+      sudo systemctl stop containerd
+
+#. Run the :file:`create_stack.sh` script to initialize the Kubernetes node
+   and setup a container network plugin.
+
+   .. code-block:: bash
+
+      sudo /usr/share/clr-k8s-examples/create_stack.sh minimal
```
Comment (eadamsintel) [inline source/tutorials/kubernetes.rst]: There has to be a better way than this.   Try adding `sudo chown -R $USER $HOME/.kube` as a step to see if it eliminates the need for sudo on kubectl. I gleamed this from https://stackoverflow.com/questions/51280172/how-to-use-kubectl-command-instead-of-sudo-kubectl#51280522

File: source/tutorials/kubernetes.rst [hunk 18]
```diff
@@ -3,354 +3,716 @@
...
+      
+      In this example, the :command:`--pod-network-cidr 10.244.0.0/16`
+      parameter is to use *flannel* as the pod networking. See `Choose a pod
+      network add-on`_ for more information.
 
-    .. code-block:: bash
+   .. code-block:: bash
 
-       exit
+      sudo kubeadm init \
```
Comment (eadamsintel) [inline source/tutorials/kubernetes.rst]: Did this actually work? I had to do the following for CRIO. I saw you had this in the containerd section. 
```
sudo mkdir -p /etc/systemd/system/kubelet.service.d/
cat << EOF | sudo tee  /etc/systemd/system/kubelet.service.d/10-cgroup-driver.conf
[Service]
Environment="KUBELET_EXTRA_ARGS=--cgroup-driver=systemd"
EOF
sudo systemctl daemon-reload
```

File: source/tutorials/kubernetes.rst (hunk 17 above)
Comment (eadamsintel) [reply source/tutorials/kubernetes.rst]: Alternatively, This is what the stack suggests to do after the stack is initialized. It might be enough to just `sudo rm -rf $HOME/.kube` and then run the following.

```

  mkdir -p $HOME/.kube
  sudo cp -i /etc/kubernetes/admin.conf $HOME/.kube/config
  sudo chown $(id -u):$(id -g) $HOME/.kube/config

```

File: source/tutorials/kubernetes.rst [hunk 19]
```diff
@@ -3,354 +3,716 @@
...
+   .. code-block:: bash
+      
+      sudo systemctl daemon-reload
+
+
+If you had a previously failed initialization due to a proxy issue, restart
+the process with the :command:`kubeadm reset` command.
+
+
+DNS issues
```
Comment (eadamsintel) [inline source/tutorials/kubernetes.rst]: I found a few more error conditions and how to solve. These need to be wordsmith ed and extraneous details removed. 

1. coredns pods stuck in container creating.  Logs show something like the following.
```
  Warning  FailedCreatePodSandBox  5m7s                 kubelet, kata3     Failed to create pod sandbox: rpc error: code = Unknown desc = failed to get network JSON for pod sandbox k8s_coredns-6955765f44-5gpj2_kube-system_9ff6ddc2-189f-40fc-bd48-8a392dfdc744_0(ea3135094c88babb982f45de17fe5716a3320dd2eb79990c5eb8959c11cfc680): cannot convert version ["" "0.1.0" "0.2.0"] to 0.4.0
```
Soln: In this case the /etc/cni/net.d/10-flannel.conflist or another CNI file is using an incompatible (probably older) version. Delete the file and restart the stack. 

2. coredns pod stuck in container creating. Logs show something like the following.
```
  Warning  FailedCreatePodSandBox  117s (x197 over 45m)  kubelet, kata3     (combined from similar events): Failed to create pod sandbox: rpc error: code = Unknown desc = failed to create pod network sandbox k8s_coredns-6955765f44-npsm5_kube-system_323177a2-2577-4748-96ed-dd8c265d8ad4_0(98b34e90afd36c948c171977248313d5e378a8e1926a14377d922cb22db9f104): error getting ClusterInformation: Get https://[10.96.0.1]:443/apis/crd.projectcalico.org/v1/clusterinformations/default: x509: certificate signed by unknown authority (possibly because of "crypto/rsa: verification error" while trying to verify candidate authority certificate "kubernetes")
```
soln: There are probably multiple CNI configuration files in /etc/cni/net.d  Delete them all and start again.

3. coredns pods stuck in pending
```
Warning  FailedScheduling  55s (x3 over 2m12s)  default-scheduler  0/1 nodes are available: 1 node(s) had taints that the pod didn't tolerate.
```
soln 2.1 - If /etc/cni/net.d is blank then try applying a CNI plugin. It should populate that directory with a CNI configuration

**Comment by mvincerx** [comment]
Eric will contact Miguel.

File: source/tutorials/kubernetes.rst [hunk 20]
```diff
@@ -3,354 +3,724 @@
...
 
-#. Switch to root to modify the `hosts` file:
+See `What is a Container Runtime Interface (CRI)?`_ for more information on
+what a CRI is.
+
+|CL| supports Kubernetes with the various runtimes
+below with or without `Kata Containers`_:
+
+* `CRI+O`_
+* `containerd`_
```
Comment (eadamsintel) [inline source/tutorials/kubernetes.rst:260]: I've seen a lot of sites capitalize containerd when it stands alone and have seen other sites not capitalize. I don't know which way is correct.

File: source/tutorials/kubernetes.rst [hunk 21]
```diff
@@ -3,354 +3,724 @@
...
 
-Proxy configuration (optional)
-******************************
+      sudo kubeadm init \
+      --pod-network-cidr 10.244.0.0/16 
 
-If you use a proxy server, you must set your proxy environment variables and create an appropriate proxy configuration file for both CRI-O services. Consult your IT department if you are behind a corporate proxy for the appropriate values. Ensure that your local IP is **explicitly included** in the environment variable *NO_PROXY*. (Setting *localhost* is not enough.)
 
-If you have already set your proxy environment variables, run the following commands as a shell script to configure all of these services in one step:
+#. (Optional) Docker can be set to use the kata-runtime as the default Docker
```
Comment (eadamsintel) [inline source/tutorials/kubernetes.rst:448]: This is a bad idea and is not recommended.  Using Kata with Docker with Kubernetes is not a recommended setup.  Please remove any Kata references.  Once you fix a runtime in Docker you can't change it and some pods need extra privileges.  This is why CRIO or containerd are the recommended setup for using Kata.

File: source/tutorials/kubernetes.rst [hunk 22]
```diff
@@ -3,354 +3,724 @@
...
+Add the Kata runtime to Kubernetes
+-----------------------------------
+
+For information on Kata as a container runtime, see `What is Kata
+Containers?`_. Using Kata Containers is optional.
+
+You can use *kata-deploy* to install all the necessary parts of Kata
+Containers after you have a Kubernetes cluster running with one of the CRI's
+using the default runc runtime. Follow the steps in the Kubernetes quick start
+section of the  `kat-containers GitHub README
```
Comment (eadamsintel) [inline source/tutorials/kubernetes.rst:466]: It should be "kata containers"

File: source/tutorials/kubernetes.rst [hunk 23]
```diff
@@ -3,354 +3,724 @@
...
 
-  #. To ensure your system isn't running previous session variables, run:
+The default built-in runtime provided by Kubernetes is using the system Docker
+installation via Dockershim and as a result is one of the simplest to use. One
+limitation of using Dockershim is that all pods on the Kubernetes node will
+inherit and use the default runtime that Docker is set to use. To be able to
+specify a container runtime per-Kerbernetes service, use CRI+O or containerd. 
 
-     .. code-block:: bash
+What is Kata Containers?
```
Comment (eadamsintel) [inline source/tutorials/kubernetes.rst:699]: This had a different bold to the heading as compared to the other "What is ..."

**Comment by eadamsintel** [comment]
I did not try out the steps in the latest doc and only skimmed through it. I did not see what kind of CNI error messages you might get when using mismatched versions. I think that would be really helpful for someone who gets stuck at the step where the DNS pods are initializing and get stuck.

File: source/tutorials/kubernetes.rst (hunk 23 above)
Comment (puneetse) [reply source/tutorials/kubernetes.rst:699]: This is intentional. The 3 CRIs are a a subsection of the `What is a Container Runtime Interface (CRI)?`.

The `What is Kata Containers?` is not in the same level as the CRIs.

File: source/tutorials/kubernetes.rst (hunk 21 above)
Comment (puneetse) [reply source/tutorials/kubernetes.rst:448]: I've removed this step.

File: source/tutorials/kubernetes.rst (hunk 20 above)
Comment (puneetse) [reply source/tutorials/kubernetes.rst:260]: I'll leave this up to TCS in a follow up editorial PR.

File: source/tutorials/kubernetes.rst (hunk 23 above)
Comment (eadamsintel) [reply source/tutorials/kubernetes.rst:699]: Ah ok.  It is hard to read without all the proper 01.org formatting that would go around it.

File: source/tutorials/kubernetes.rst (hunk 16 above)
Comment (mvincerx) [reply source/tutorials/kubernetes.rst:88]: 👍  Good call. I suggest making this a note.

File: source/tutorials/kubernetes.rst (hunk 20 above)
Comment (mvincerx) [reply source/tutorials/kubernetes.rst:260]: It's fine as is. CNCF defines this: https://containerd.io/

File: source/tutorials/kubernetes.rst (hunk 19 above)
Comment (puneetse) [reply source/tutorials/kubernetes.rst:606]: I added these to the PR.

File: source/tutorials/kubernetes.rst (hunk 16 above)
Comment (puneetse) [reply source/tutorials/kubernetes.rst:88]: I added this information.

File: source/tutorials/kubernetes.rst (hunk 17 above)
Comment (puneetse) [reply source/tutorials/kubernetes.rst:119]: I'm going to leave this for now because I have observed some inconsistency without `sudo` in the past. Launching with `sudo` should not impact anything since subsequent `sudo` command run. It also makes the whole section consistent with `sudo` prefixed before each command.

File: source/tutorials/kubernetes.rst (hunk 23 above)
Comment (puneetse) [reply source/tutorials/kubernetes.rst:731]: You should pull the fork and build it locally :)

File: source/tutorials/kubernetes.rst (hunk 15 above)
Comment (puneetse) [reply source/tutorials/kubernetes.rst:169]: We switched to kata-deploy as discussed. Let me know if there is anything else you want to add.

**Comment by mvincerx** [comment]
> I did not try out the steps in the latest doc and only skimmed through it. I did not see what kind of CNI error messages you might get when using mismatched versions. I think that would be really helpful for someone who gets stuck at the step where the DNS pods are initializing and get stuck.

@eadamsintel Could you test this? 

I'd like to test the automated setup script, at minimum. Plan to do so today after checking on NUC.

**Comment by mvincerx** [comment]
@ttzeng or @odwdinc can either of you test this guide?

File: source/tutorials/kubernetes.rst [hunk 24]
```diff
@@ -3,354 +3,756 @@
...
 *************
 
 This tutorial assumes you have already installed |CL|. For detailed
-instructions on installing |CL| on a bare metal system, follow the
-:ref:`bare metal installation tutorial<bare-metal-install-desktop>`.
-Learn about the benefits of having an up-to-date system for cloud
-orchestration on the :ref:`swupd-guide` page.
+instructions on installing |CL| on a bare metal system, follow the :ref:`bare
+metal installation tutorial<bare-metal-install-desktop>`.
+
```
[the remark this answers is not in the document]
Comment (mvincerx) [reply source/tutorials/kubernetes.rst:46]: Ln 47 > work for me. Thanks.

**Comment by mvincerx** [comment]
@eadamsintel, Waiting on your feebdack or approval.

**Comment by eadamsintel** [comment]
LGTM.

**Comment by mvincerx** [comment]
> LGTM.
> Please add a keyword to close at least 1 issue in a commit: Closes #1003
> If you plan to keep on improving the k8s tutorial, it's fine to leave open #894.

@puneetse. This PR needs to close at least one issue, yet it currently lacks a keyword to do so.
Will you please respond to my question above? IMHO: This PR should close both issues. You can always open a new issue, based on this PR's *latest* revisions.

File: source/tutorials/kubernetes.rst [hunk 25]
```diff
@@ -3,354 +3,761 @@
...
+      *flannel* for pod networking (see `Choose a pod network add-on`_), the
+      :file:`/etc/crio/crio.conf` file needs to include the value below. On
+      |CL| this is done automatically. 
 
-       sudo systemctl enable crio.service
+      .. code-block:: console
 
-#.  Enter these commands to restart the system services so that CRI-O will be restarted, and will automatically start with every reboot.
+         [crio.runtime]
+         manage_network_ns_lifecycle = true
```
Comment (mythi) [inline source/tutorials/kubernetes.rst:327]: is is the default setting so no longer needed in this doc afaics: https://github.com/clearlinux-pkgs/cri-o/blob/master/make_args#L7

File: source/tutorials/kubernetes.rst [hunk 26]
```diff
@@ -3,354 +3,761 @@
...
+
+For information on Kata as a container runtime, see `What is Kata
+Containers?`_. Using Kata Containers is optional.
+
+You can use *kata-deploy* to install all the necessary parts of Kata
+Containers after you have a Kubernetes cluster running with one of the CRI's
+using the default runc runtime. Follow the steps in the Kubernetes quick start
+section of the  `kata-containers GitHub README
+<https://github.com/kata-containers/packaging/tree/master/kata-deploy#kubernetes-quick-start>`_
+to install Kata.
```
Comment (mythi) [inline source/tutorials/kubernetes.rst:474]: Kata runtime is already part of `cloud-native-basic` bundle so this is not needed. Only the part that adds ? 
`RuntimeClass` settings is needed.

File: source/tutorials/kubernetes.rst [hunk 27]
```diff
@@ -3,354 +3,761 @@
...
 
-    .. code-block:: bash
 
-       sudo systemctl daemon-reload
-       sudo systemctl restart crio
+#. Once the cluster initialization is complete, continue reading about how to
+   `Use your cluster`_.
 
 
+containerd
```
Comment (mythi) [inline source/tutorials/kubernetes.rst:334]: You don't specify how to deal with cni-plugin dirs for containerd. cri-o knows many search paths and we've configured it to search `/opt/cni/bin`  and `/usr/libexec/cni`. `/opt/cni/bin` is not enough unless you tell users to install cni-plugins (what Clear ships in `/usr/libexec/cni`) there too.
